compiler/rustc_query_system/src/dep_graph/graph.rs

```diff
@@ -117,16 +119,23 @@ impl<K: DepKind> DepGraph<K> {
     ) -> DepGraph<K> {
         let prev_graph_node_count = prev_graph.node_count();
 
+        let current =
+            CurrentDepGraph::new(prev_graph_node_count, encoder, record_graph, record_stats);
+
+        // Instantiate an *always green* node for dependency-less anonymous queries.
```

Comment:
I would not bring the fact that dependency-less nodes are always green into play here because it actually does not matter for the optimization, right?